.github/ISSUE_TEMPLATE/spec_request.md

```diff
@@ -0,0 +1,55 @@
+---
+title: 'Request creation of a new format specification'
+labels: spec-request
+---
+<!--
+A request for a format specification consists of 2 YAML blocks followed by a free-form text description.
+Fill in all the fields possible. Delete fields you cannot populate and all the comments from this template (including this one!).
+The resulting YAML blocks should contain valid YAML.
+Mandatory fields are marked with the word MANDATORY for your convenience.
+-->
+```yaml
+# This block is the boilerplate of a KSY spec file. It must be a valid KSY. It is meant to be ready to be copied as it is in order to make the task of the implementer easier.
+meta: # see https://doc.kaitai.io/ksy_reference.html#meta for more info
+  id: # insert an ID you feel right to assign to the format here. KS currently has no namespaces, so the id MUST be unique within all the repo and you should be sure there will be no collisions. If in doubt, add a prefix of an app name. MANDATORY.
+  title: # insert a human-readable name of the format here. MANDATORY.
+  application: # insert a human-readable name of the application for which this format is native here. If it is native for several applications, choose the ones you feel the most affiliated with the format. If the format is a widespread one, like PNG or WAV, just remove this field.
+  file-extension:
+    # insert here a YAML list of file extensions should be placed here.
+  xref: # check if the following sources contain info about the format you request. See https://doc.kaitai.io/user_guide.html#meta for more info
+    mime: # MIME type
+    rfc: # Link to RFC
+    din:
+      id: # DIN standard identifier
+    iso: # ISO standard identifier
+    forensicswiki: # Page name on https://forensicswiki.org/
+    justsolve: # ID on http://fileformats.archiveteam.org/wiki/
+    pronom: # IDs on https://www.nationalarchives.gov.uk/pronom/fmt/
+     - x-fmt/ 
+    wikidata: Q # ID on WikiData
+    fileformat: # Link to https://www.fileformat.info
+    loc: # ID in https://www.loc.gov/preservation/digital/formats/    din:
+      id: # DIN standard identifier
+
+doc: |
+  # Describe your format briefly AND add links to the sources of sample files in the Web. MANDATORY
+
+doc-ref:
+  # A YAML array of URIs to information sources and/or their names goes here. MANDATORY. See https://doc.kaitai.io/ksy_reference.html#doc-ref for more info on sources which are not just an URI
+  # An effective strategy may be to search specifically for PDFs (filetype:pdf in Google) and source code on GitHub (links to source code files are especially valuable for understanding poorly documented parts) or directly in the repos of free software using the format.
+  # Prefer sources with a public-domain-like licenses over permissive ones, permissive ones over copyleft ones and free licenses over proprietary ones
```

Comment:
Absolutely not, this is not how it works. By definition, we should prefer official sources first of all, and only if they're bad in some way (non-free, non-redistributable, too vague, badly structured, etc) — we should bring in alternative ones (i.e. reverse engineered, unofficial specifications / implementations, etc).

Reply:
>By definition, we should prefer official sources first of all, and only if they're bad in some way (non-free, non-redistributable, too vague, badly structured, etc) — we should bring in alternative ones (i.e. reverse engineered, unofficial specifications / implementations, etc).

Official sources are often non-free, which means we cannot copy the quotes from them. Without `doc`s IMHO complex specs are incomplete - they carry info on how to parse but not the info what each field means. I mean they cannot be replaced with the spec entirely, but it is the goal because it is easier even for a human to read specs rather than free-form text. I mean, if we have a KSY spec with all the needed `doc`s, we don't need the official spec anymore.

Reply:
I think in most cases you should just include all reference material that you can find, official or not. From my experience you can never have too many reference documents for a format, in case one of them has ambiguous or imprecise information. Unless you have literally tens/hundreds of reference documents, there's no need to be picky.

In any case, I think we should always use both official and unofficial reference material if at all possible. Official docs tell you exactly how the format is *supposed* to work and what is and isn't valid, but might not accurately reflect how the format is implemented and used in reality. Unofficial docs tell you more about how the format is actually used in practice and what oddities there might be in certain real-world implementations, but they are sometimes incomplete or wrong (especially when reverse-engineered). By cross-referencing both official and unofficial docs you get a much more accurate spec than by only using one or the other.

Reply:
I mostly aggree, but JMHO 5 links with the same info are too many. Could you review the new text, addressing the issues you have mentioned?